spacepy/irbempy/__init__.py

```diff
@@ -83,11 +83,2467 @@
+    else:
+        results['LCDS'] = dm.dmfilled([nTAI, ])
+        results['K'] = dm.dmfilled([nTAI, ])
+    results['LCDS'].attrs['DESCRIPTION'] = " ".join(["Last closed drift shell",
+                                                     "calculated with SpacePy's",
+                                                     "irbempy module"])
+    results['LCDS'].attrs['UNITS'] = "dimensionless"
+    results['LCDS'].attrs['DEPEND_0'] = "UTC"
+    results['LCDS'].attrs['DEPEND_1'] = "K"
+    results['LCDS'].attrs['MODEL'] = "{0} (IRBEM implementation)".format(extMag)
+    results['K'].attrs['DESCRIPTION'] = "Modified 2nd adiabatic invariant at LCDS"
+    results['K'].attrs['UNITS'] = "R_E.G^{1/2}"
+    results['K'].attrs['DEPEND_0'] = "UTC"
+    results['K'].attrs['DEPEND_1'] = "AlphaEq"
+    results['K'].attrs['MODEL'] = "{0} (IRBEM implementation)".format(extMag)
+    results['UTC'] = ticks.UTC
+    results['AlphaEq'] = dm.dmarray(alpha,
+                                    attrs={'DESCRIPTION': ' '.join(['Equatorial pitch',
+                                                                    'angle for LCDS',
+                                                                    'calculation']),
+                                           'UNITS': 'degrees'})
+
+    mlt *= 15  # hours to degrees
+    mlt = np.deg2rad(mlt)
+
+    options = (int4 * 5)(*options)
+
+    for idxt, tt in enumerate(ticks):
+        if not omnivals:
+            # prep_irbem will get omni if not specified, but to save on repeated calls, do it once here
+            import spacepy.omni as omni
+            omnivals = omni.get_omni(tt)
+        for idxa, pa in enumerate(alpha):
+            b1x = -1.0*bracket[0]*np.cos(mlt)
+            b1y = -1.0*bracket[0]*np.sin(mlt)
+            loci_brac1 = spc.Coords([b1x, b1y, 0], 'GSM', 'car', use_irbem=True)
+            if 'verbose' in kwargs:
+                print('Initial inner bracket: {0}'.format(loci_brac1))
+
+            d = prep_irbem(tt, loci_brac1, alpha=[pa], extMag=extMag, options=options, omnivals=omnivals)
+            badval = d['badval']
+            kext = d['kext']
+            sysaxes = d['sysaxes']
+            iyearsat = d['iyearsat']
+            idoysat = d['idoysat']
+            secs = d['utsat']
+            xin1 = d['xin1']
+            xin2 = d['xin2']
+            xin3 = d['xin3']
+            magin = d['magin']
+            nTtoG = 1.0e-5
+
+            bmin = np.empty((), np.float64)
+            GEOcoord = np.empty((3,), np.float64)
+            irbemlib.find_magequator1(
+                int4(kext), options, int4(sysaxes), int4(iyearsat[0]),
+                int4(idoysat[0]), real8(secs[0]), real8(xin1[0]),
+                real8(xin2[0]), real8(xin3[0]),
+                magin[:, 0].ctypes.data_as(ctypes.POINTER(real8 * 25)),
+                bmin.ctypes.data_as(ctypes.POINTER(real8)),
+                GEOcoord.ctypes.data_as(ctypes.POINTER(real8 * 3))
+            )
```

Comment:
I think this whole block is done atleast 2-3 times, with the same code, I am not positive it would make it cleaner to make a common method to do this(or possible without big changes even), it's just something that caught my eye when looking it over.

Reply:
The whole sequence from unpacking a prep_irbem output through setting bad values to nan seems worth making a nested function. I think we can say that crosses the line to out of scope for this PR--it's a rework of how irbempy is structured. Captured as #768 .